tests/rest/client/test_room_access_rules.py

```diff
@@ -482,6 +463,14 @@ def test_unrestricted(self):
             expect_code=403,
         )
 
+        # We can't publish the room to the public rooms directory
+        url = "/_matrix/client/r0/directory/list/room/%s" % self.unrestricted_room
+        data = {"visibility": "public"}
+
+        request, channel = self.make_request("PUT", url, data, access_token=self.tok)
+        self.render(request)
+        self.assertEqual(channel.code, 403, channel.result)
```

Comment:
This needs to be in the tests for `restricted` and `direct` as well (adapting it accordingly)